clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/handlers/TitusDeployHandler.groovy

```diff
@@ -290,8 +290,28 @@ class TitusDeployHandler implements DeployHandler<TitusDeployDescription> {
         }
       }
 
-      task.updateStatus BASE_PHASE, "Submitting job request to Titus..."
-      String jobUri = titusClient.submitJob(submitJobRequest)
+      String nextServerGroupName = resolveJobName(description, submitJobRequest, task, titusClient)
+      String jobUri
+
+      retrySupport.retry({
+        try {
+          task.updateStatus BASE_PHASE, "Submitting job request to Titus..."
+          jobUri = titusClient.submitJob(submitJobRequest)
+        } catch (io.grpc.StatusRuntimeException e) {
+          task.updateStatus BASE_PHASE, "Error encountered submitting job request to Titus ${e.status} ${e.message}"
+          if (e.status == Status.RESOURCE_EXHAUSTED || (e.status == Status.INVALID_ARGUMENT)) {
+            nextServerGroupName = resolveJobName(description, submitJobRequest, task, titusClient)
+            throw e;
+          }
+          if (e.status == Status.UNAVAILABLE) {
+            throw e;
+          }
+        }
+      }, 6, 1000, true)
```

Comment:
do you want to do exponential here? 

as written this would be 6 requests over ~30s ... is that better than 30 or 15 requests with 1s or 2s in between?